contentcuration/contentcuration/frontend/channelList/views/Channel/ChannelList.vue

```diff
@@ -97,20 +101,13 @@
         return this.listType === ListTypes.EDITABLE;
       },
     },
-    beforeRouteEnter(to, from, next) {
-      if (listTypeValidator(to.params.listType)) {
```

Comment:
Any reason for removing the defensive guards here? This will prevent us from dealing with doing bad loads for non-existent list types, as it will just prevent the navigation in the first place.

Reply:
Oh yes, that's not needed anymore and `listType` is no more available in `to.params`.

If you have a look [at the router structure](https://github.com/learningequality/studio/blob/d7c0c40d6f79d7e18efeb1a5a87edfa570a8ee26/contentcuration/contentcuration/frontend/channelList/router.js), you'll see that bad-loads will fall into [catch-all](https://github.com/learningequality/studio/blob/d7c0c40d6f79d7e18efeb1a5a87edfa570a8ee26/contentcuration/contentcuration/frontend/channelList/router.js#L73) route and will be automatically redirected to edit tab. The [validator is still](https://github.com/learningequality/studio/pull/2024/files/d7c0c40d6f79d7e18efeb1a5a87edfa570a8ee26#diff-0e636aea80644ee5a251d04cc270dfc2R76) there and now we receive list type only through `props` so there's no need for this in-component route coupling.

This also gives us the freedom to choose any route URLs [as Devon pointed out](https://github.com/learningequality/studio/pull/2024#discussion_r465137918).